app/controllers/projects_controller.rb

```diff
@@ -61,11 +61,30 @@ def update # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
 
   def activity
     authorize! @project
-    # No necessary code here
+
+    project_activities = @project.namespace.retrieve_project_activity.order(created_at: :desc)
+
+    @pagy, raw_activities = pagy(project_activities, limit: 10)
+
+    @activities = @project.namespace.human_readable_activity(raw_activities)
+
+    respond_to do |format|
+      format.html
+      format.turbo_stream
+    end
   end
 
-  def transfer
+  def transfer # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
     if Projects::TransferService.new(@project, current_user).execute(new_namespace)
+
+      @project.namespace.create_activity key: 'namespaces_project_namespace.transfer', owner: current_user,
```

Comment:
**nitpick (blocking):** Can you remove the addition of the activity here since you do this in `app/services/projects/transfer_service.rb`

Reply:
Removed in 032cb1c08ee993dee5d18b74ef198b03bafe07b3